### PR TITLE
add Chinese translation with index pages

### PR DESCRIPTION
**Pull request by rootsongjc**
This is the first PR for Chinese translation of istio docs. @geeknoid 
I had translated the index pages for:
- homepage
- blog
- docs
- about
- community
- help
- search

@fleeto can you review it for me?
Hope more and more servicemesher in China can take part in this activity.

### CODE REVIEW

**Comment by geeknoid** [comment]
Deploy preview for *preliminary-istio* failed.

Built with commit 084c1f7d827dade4554dc7aa49738ac28403b22a

https://app.netlify.com/sites/preliminary-istio/deploys/5b1f5bbf1f12b740b62c8aba

**Comment by rootsongjc** [comment]
@gyliu513 @geeknoid I add English language config in hugo config file and build successfully in local.

File: content_cn/about/_index.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+title: 关于Istio
+linktitle: 关于
+description: 关于Istio的说明
```
Comment (gyliu513) [inline content_cn/about/_index.md]: 末尾加个句号

File: content_cn/about/_index.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+weight: 15
+---
+
+了解关于Istio项目的详细信息。
+
+- [Istio是什么？](/about/intro/) 了解Istio设计要解决的问题。
+
+- [发布说明](/about/notes/)。了解最新功能、改进和错误修复。
+
+- [构建发布节奏](/about/release-cadence/)。了解我们如何管理、编号并支持Istio发布。
```
Comment (gyliu513) [inline content_cn/about/_index.md]: s/编号/编译？

Comment (rootsongjc) [reply content_cn/about/_index.md]: 原文是`number`，就是给发布编号命名吧。

Comment (gyliu513) [reply content_cn/about/_index.md]: image

我的文档没有这一小节？

Comment (rootsongjc) [reply content_cn/about/_index.md]: 这个在仓库代码里有，不过在web页面上没有。见 https://github.com/istio/istio.github.io/blob/master/content/about/_index.md

Comment (gyliu513) [reply content_cn/about/_index.md]: 明白，现在1.0在web页面不显示了。

**Comment by rootsongjc** [comment]
@gyliu513 Do you know why does this deployment fail?

**Comment by geeknoid** [comment]
It's failing due to style checking errors:

11:27:32 PM: ./content_cn/_index.md:65: MD009 Trailing spaces
11:27:32 PM: ./content_cn/_index.md:80: MD009 Trailing spaces
11:27:32 PM: ./content_cn/_index.md:95: MD009 Trailing spaces
11:27:32 PM: ./content_cn/_index.md:110: MD009 Trailing spaces
11:27:32 PM: ./content_cn/_index.md:118: MD009 Trailing spaces
11:27:32 PM: ./content_cn/_index.md:126: MD009 Trailing spaces
11:27:32 PM: ./content_cn/_index.md:131: MD009 Trailing spaces
11:27:32 PM: ./content_cn/_index.md:135: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:16: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:21: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:25: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:29: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:33: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:38: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:42: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:46: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:50: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:54: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:58: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:62: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:66: MD009 Trailing spaces
11:27:32 PM: ./content_cn/search/_index.md:70: MD009 Trailing spaces
11:27:32 PM: ./content_cn/community/_index.md:22: MD009 Trailing spaces
11:27:32 PM: ./content_cn/community/_index.md:29: MD009 Trailing spaces
11:27:32 PM: ./content_cn/community/_index.md:36: MD009 Trailing spaces
11:27:32 PM: ./content_cn/community/_index.md:43: MD009 Trailing spaces
11:27:32 PM: ./content_cn/community/_index.md:50: MD009 Trailing spaces
11:27:32 PM: ./content_cn/community/_index.md:60: MD009 Trailing spaces

**Comment by geeknoid** [comment]
Right now, we spell check the english content. I don't know if our spell checker supports Chinese, but it's worth trying it out. See the Makefile in the root for the spell check commands we use.

**Comment by geeknoid** [comment]
These pages have a lot of special formatting stuff not found in other pages (scripts, styles, etc). We should figure out how to share this stuff instead of having it repeated in this way. Go ahead with this change, but I'm going to try to move this stuff out of the content directories and into the infrastructure instead.

**Comment by rootsongjc** [comment]
I think we should create a markdown rule for Chinese content. The lint command `mdl --ignore-front-matter --style mdl_style.rb .` check the markdown style with `mdl_style.rb` that suit for English but not for Chinese writing habit. So could we create a `mdl_style_cn.rb` for Chinese content to get rid of these deployment errors? @geeknoid

**Comment by rootsongjc** [comment]
When running `htmlproofer` it report in netlify console:
```bash
12:22:53 PM: - ./public/cn/community/index.html
12:22:53 PM:   *  External link https://preliminary.istio.io/cn/about/ failed: 404 No error
12:22:53 PM:   *  External link https://preliminary.istio.io/cn/community/ failed: 404 No error
12:22:53 PM:   *  External link https://preliminary.istio.io/cn/docs/ failed: 404 No error
12:22:53 PM:   *  External link https://preliminary.istio.io/cn/help/ failed: 404 No error
```
Does we miss some config in Makefile?

**Comment by geeknoid** [comment]
Hmmm. Seems like this should work. The fact htmlproofer sees cn/community/index.html in the first place means that the Chinese content was output by Hugo. So why doesn't it see the about page and the others?

I'm going to have to pull down your PR and try it out locally to figure out what's going on. I can probably get to this tonight.

**Comment by geeknoid** [comment]
Oh, I know what this is about. We currently have a chicken-and-egg problem. Any PR which adds a new file and at the same time adds a link to that file will get this error.

To fix this, we'll have to break things up. We'll need linting to happen in one step, probably using CircleCi, and we'll have Netlify just publish the thing. I tried to get linting on CircleCi but ran into all sorts of install issues with the Node.js and Gem linters. They just worked trivially on Netlify, but CircleCi's environment is different. I guess I'll have to try again.

**Comment by rootsongjc** [comment]
During the building stage `htmlproofer` cannot access the `https://preliminary.istio.io/cn/*` URLs. To avoid this during building, put `/preliminary.istio.io/cn/` on the tail of `htmlproofer` comman in the Makefile.

**Comment by geeknoid** [comment]
Regarding the style checking, I think it would be fine to have per-language style checking rules as needed.

Regarding adding preliminary.istio.io/cn to the htmlproofer exception list, I think that would not be a good idea as that would prevent link checking on any link pointing to the chinese translation. It'd be easy to introduce broken links that way.

**Comment by gyliu513** [comment]
@geeknoid why disabled showing 1.0 on the dashboard?

**Comment by geeknoid** [comment]
Could you rephrase that, I don't understand the question.

**Comment by rootsongjc** [comment]
@gyliu513 @geeknoid Does the docs 1.0 will show until istio 1.0 release?

**Comment by geeknoid** [comment]
preliminary.istio.io shows the content of the master branch of
istio.github.io. This is currently configured to show Istio Preliminary 1.0
and will stay that way until 1.0 is released.

istio.io shows the content of the release-0.8 branch of istio.github.io and
will stay that way until 1.0 is released.

When 1.0 is released, we will create a release-1.0 branch and have istio.io
display that branch. preliminary.istio.io will then be used for 1.1 work.

**Comment by rootsongjc** [comment]
@geeknoid One more question, we have to hard code the absolute path to root in the Chinese contents, for example:
- en: `/docs/tasks`
- cn: `/cn/docs/tasks`

As we know all the links in `content_cn` should add  a `/cn` prefix but now we have to hard code these, is there a good method to give a relative language url in Chinese contents?

**Comment by geeknoid** [comment]
I think we can play some games with the baseURI to force Hugo to do the right thing with relative paths. I'll have to experiment a bit. It's tricky because of the many hosting environments for the site (preliminary.istio.io, istio.io, archive.istio.io, localhost, PR previews)

I checked in a few changes around translation support:

- I fixed several bugs so the site behaves better now, sets the lang tag, etc.

- I added a menu selection to let the user pick the language to view. This currently just redirects the user to the appropriate home page. Eventually, I want to make it store a cookie so the browser will remember where to go.

- I renamed content_cn to content_zh. "zh" is the language code for Chinese, "cn" is the country code for China. We want to use using language codes here.

- I tried out a redirect on Netlify and it works great. So when you're ready with the full translation, we can make it so Chinese users automatically go to the Chinese version of the site, unless the cookie is set.

**Comment by rootsongjc** [comment]
That's great. Hope to use the relative path in markdown. Maybe hugo's [absLangURL](https://gohugo.io/functions/abslangurl/) can help.
